### PR TITLE
Fix param binding issues

### PR DESCRIPTION
**Pull request by RDPerera**
## Purpose

Fixes:

1. Add union-type support for import params in RFC client
2. Add nill support for export params
3. Add Date support for Import params
4. Add byte array support for Import params
5. Fix version binding.

## Checklist

- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java [hunk 1]
```diff
@@ -51,11 +51,15 @@ public boolean supportsEvents() {
...
-        destinationProperties.put(destinationName.toString(), properties);
+        try {
+            jcoDestinationConfig.entrySet().forEach(entry -> {
+                BString key = entry.getKey();
+                BString value = (BString) entry.getValue();
+                properties.setProperty(SAPConstants.CONFIG_KEYS.get(key.toString()), value.toString());
+            });
+            destinationProperties.put(destinationName.toString(), properties);
+        } catch (Exception e) {
+            throw new RuntimeException("Error while adding destination: " + e.getMessage());
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java:62]:
```suggestion
            throw new RuntimeException("Error while processing destination properties: " + e.getMessage());
```

Comment (niveathika) [reply native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java:62]: Add e also to the error

File: native/src/main/java/io/ballerina/lib/sap/Client.java [hunk 2]
```diff
@@ -183,13 +212,52 @@ private static void setImportParams(JCoParameterList jcoPramList, BMap<BString,
...
                             entry.getKey().toString() + ". Unsupported type " + type + " Supported types are: " +
                             "string, int, float, decimal", null);
             }
         });
     }
-
+    private static void throwUnsupportedUnionTypeError(Object key) {
+        throw SAPErrorCreator.fromBError("Error while setting input parameter for field: " +
+                key.toString() + ". Unsupported union type. Supported types are: string, int, float, decimal", null);
+    }
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/lib/sap/Client.java:262]:
```suggestion
        throw SAPErrorCreator.fromBError("Unsupported union type '" + + "' for field '" +
                key.toString() + '". Supported types are: string, int, float, decimal and nil able supported types. ", null);
    }
```

Comment (niveathika) [reply native/src/main/java/io/ballerina/lib/sap/Client.java:262]: Add the coming type in the error message as well.

Comment (RDPerera) [reply native/src/main/java/io/ballerina/lib/sap/Client.java]: Changed with [010b6b2](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/12/commits/010b6b2c237f70b1e7db75e29c8c780db22968de).

File: native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java (hunk 1 above)
Comment (RDPerera) [reply native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java:62]: Changed with [010b6b2](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/12/commits/010b6b2c237f70b1e7db75e29c8c780db22968de).

**Review by niveathika — APPROVED** [review]

**Pull request merged**